### PR TITLE
Fix training resuming in references/segmentation #2116

### LINKED ISSUE #2116: Resume in the Segmentation example
## 🐛 Bug

The segmentation training [script](https://github.com/pytorch/vision/blob/7b60f4db9707d7afdbb87fd4e8ef6906ca014720/references/segmentation/train.py) doesn't seem to allow correct resuming. I think [lines 131-133](https://github.com/pytorch/vision/blob/7b60f4db9707d7afdbb87fd4e8ef6906ca014720/references/segmentation/train.py#L131) should be changed to:

```
    start_epoch = 0
    if args.resume:
        checkpoint = torch.load(args.resume, map_location='cpu')
        model.load_state_dict(checkpoint['model'])
        optimizer.load_state_dict(checkpoint['optimizer'])
        lr_scheduler.load_state_dict(checkpoint['lr_scheduler'])
        start_epoch = checkpoint['epoch'] + 1
```

Then, [line 161](https://github.com/pytorch/vision/blob/7b60f4db9707d7afdbb87fd4e8ef6906ca014720/references/segmentation/train.py#L161) and [lines 167-173](https://github.com/pytorch/vision/blob/7b60f4db9707d7afdbb87fd4e8ef6906ca014720/references/segmentation/train.py#L167) should be changed respectively to:

```
    for epoch in range(start_epoch, args.epochs):
```

and

```
utils.save_on_master(
            {
                'model': model_without_ddp.state_dict(),
                'optimizer': optimizer.state_dict(),
                'lr_scheduler': lr_scheduler.state_dict(),
                'epoch': epoch
            },
```

Besides, is there a good reason to set [`batch_size=1`](https://github.com/pytorch/vision/blob/7b60f4db9707d7afdbb87fd4e8ef6906ca014720/references/segmentation/train.py#L120) (instead of `args.batch_size` or even `2*args.batch_size`) for `data_loader_test`?

Thanks.

### PR DESCRIPTION
**Pull request by netw0rkf10w**
Fixed an issue discussed in #2116.

Closes #2116

### CODE REVIEW

File: references/segmentation/train.py [hunk 1]
```diff
@@ -157,8 +153,15 @@ def main(args):
         optimizer,
         lambda x: (1 - x / (len(data_loader) * args.epochs)) ** 0.9)
 
+    if args.resume:
+        checkpoint = torch.load(args.resume, map_location='cpu')
+        model_without_ddp.load_state_dict(checkpoint['model'])
```
Comment (vincentqb) [inline references/segmentation/train.py:158]: @fmassa -- I haven't tested that moving the moment that `load_state_dict` will be loaded still works with DDP. Do you want to comment on this?

**Comment by netw0rkf10w** [comment]
@vincentqb Sorry there was some space issue raised by `python_lint`. I was about to push a fix but you were too fast to approve the changes :P Do you want me to create a new PR?

**Comment by netw0rkf10w** [comment]
@vincentqb Never mind, I didn't know that I could update the PR by simply pushing to my master branch!

File: references/segmentation/train.py (hunk 1 above)
Comment (fmassa) [reply references/segmentation/train.py:158]: Great question.

TL;DR: yes, this works with DDP in the current setup, and is what we do for the classification and detection reference scripts as well, see https://github.com/pytorch/vision/blob/1affa2e8d8ffdf4f6e340f9f29de2ff06b937db7/references/classification/train.py#L191-L196 and https://github.com/pytorch/vision/blob/1affa2e8d8ffdf4f6e340f9f29de2ff06b937db7/references/detection/train.py#L112-L117

The reason why this works for DDP is that we only use 1 GPU per python process. In this case, the same code gets executed, meaning that all the GPUs will load the checkpoint and get updated accordingly. This wouldn't be the case if we were using 2 GPUs per python process (à la `DataParallel` for example), so this is important to keep in mind.
